auslib/db.py

```diff
@@ -83,6 +83,43 @@ class ChangeScheduledError(SQLAlchemyError):
     for data consistency reasons."""
 
 
+class JSONColumn(sqlalchemy.types.TypeDecorator):
+    """JSONColumns are used for types that are deserialized JSON (usually
+    dicts) in memory, but need to be serialized to text before storage.
+    JSONColumn handles the conversion both ways, serialized just before
+    storage, and deserialized just after retrieval."""
+
+    impl = Text
+
+    def process_bind_param(self, value, dialect):
+        if value:
+            value = json.dumps(value)
+        return value
+
+    def process_result_value(self, value, dialect):
+        if value:
+            value = json.loads(value)
+        return value
+
+
+def BlobColumn(impl=Text):
```

Comment:
Hmmm, I think this is more of a Column Factory than a decorator (at least, I'm not invoking it as a decorator further down). In any case, the reason is because impl in BlobColumn needs to be configurable at runtime based on the database dialect. JSONColumn is different because it uses Text as the impl regardless of dialect.